app/labor/rate_limit_checker.rb

```diff
@@ -48,13 +39,26 @@ def ping_admins
     RateLimitCheckerWorker.perform_async(user.id, action)
   end
 
+  def check_comment_creation_limit
+    user.comments.where("created_at > ?", 30.seconds.ago).size > 9
+  end
+
+  def check_published_article_creation_limit
+    user.articles.published.where("created_at > ?", 30.seconds.ago).size > 9
+  end
+
+  def check_image_upload_limit
+    Rails.cache.read("#{user.id}_image_upload").to_i > 9
+  end
+
+  def check_follow_account_limit
+    user_today_follow_count > self.class.daily_account_follow_limit
+  end
+
   private
 
   def user_today_follow_count
     following_users_count = user.following_users_count
     return following_users_count if following_users_count < self.class.daily_account_follow_limit
```

Comment:
I'm not sure why `daily_account_follow_limit` was added as a class method. We are only using it in one place except for this class and its spec:

```ruby
      # spec/requests/follows_create_spec.rb:23
      allow(rate_limit_checker).
        to receive(:user_today_follow_count).
        and_return(RateLimitChecker.daily_account_follow_limit + 1)
```

Given that we could just as well just stub the call on `SiteConfig` directly, I'm not sure it makes sense to keep this class method around at all, we could just inline the `SiteConfig.rate_limit_follow_count_daily` calls and remove the indirection.

Reply:
I agree